backend/de.metas.business/src/main/java/de/metas/bpartner/impexp/BPartnerBankAccountImportHelper.java

```diff
@@ -93,8 +100,101 @@ else if (!Check.isEmpty(importRecord.getSwiftCode(), true) && !Check.isEmpty(imp
 
 			importRecord.setC_BP_BankAccount_ID(bankAccount.getC_BP_BankAccount_ID());
 		}
+		else if (Check.isNotBlank(importRecord.getBankDetails()))
+		{
+			bankAccount = initializeBankAccountFromBankDetails(importRecord.getBankDetails());
+		}
+
+		if (Check.isNotBlank(importRecord.getIBAN()))
+		{
+			if (bankAccount == null)
+			{
+				final BankAccountQuery queryWithIban = BankAccountQuery.builder().iban(importRecord.getIBAN()).build();
+				bankAccount = handleBPBankAccountCollection(bankAccountDAO.getBpartnerBankAccount(queryWithIban));
+			}
+
+			bankAccount.setIBAN(importRecord.getIBAN());
+		}
+
+		if (Check.isNotBlank(importRecord.getQR_IBAN()))
+		{
+			if (bankAccount == null)
+			{
+				final BankAccountQuery queryWithQRIban = BankAccountQuery.builder().qrIban(importRecord.getQR_IBAN()).build();
+				bankAccount = handleBPBankAccountCollection(bankAccountDAO.getBpartnerBankAccount(queryWithQRIban));
+			}
+
+			bankAccount.setQR_IBAN(importRecord.getQR_IBAN());
+		}
+
+		if (bankAccount == null)
+		{
+			return;
+		}
+
+		if (bankAccount.getC_Currency_ID() <= 0 && Check.isBlank(importRecord.getISO_Code()))
+		{
+			throw new AdempiereException("Missing mandatory value for currency on business partner bank account!");
+		}
+
+		final CurrencyCode currencyCode = CurrencyCode.ofThreeLetterCode(importRecord.getISO_Code());
+		final CurrencyId currencyId = currencyBL.getByCurrencyCode(currencyCode).getId();
+		bankAccount.setC_Currency_ID(currencyId.getRepoId());
+
+		bankAccount.setC_BPartner_ID(bpartnerId.getRepoId());
+		bankAccount.setAD_Org_ID(importRecord.getAD_Org_ID());
+
+		InterfaceWrapperHelper.saveRecord(bankAccount);
+
+		importRecord.setC_BP_BankAccount_ID(bankAccount.getC_BP_BankAccount_ID());
+	}
+
+	@NonNull
+	private I_C_BP_BankAccount initializeBankAccountFromBankDetails(@NonNull final String bankDetails)
+	{
+		final I_C_BP_BankAccount bankAccount;
+
+		final Optional<BankId> bankIdOpt = resolveBankIdFromBankDetails(bankDetails);
+
+		if (bankIdOpt.isPresent())
+		{
+			final BankAccountQuery queryWithBankId = BankAccountQuery.builder().description(bankDetails).build();
+			bankAccount = handleBPBankAccountCollection(bankAccountDAO.getBpartnerBankAccount(queryWithBankId));
+			bankAccount.setC_Bank_ID(BankId.optionalToRepoId(bankIdOpt));
+		}
+		else
+		{
+			final BankAccountQuery queryWithBankDetails = BankAccountQuery.builder().description(bankDetails).build();
+			bankAccount = handleBPBankAccountCollection(bankAccountDAO.getBpartnerBankAccount(queryWithBankDetails));
+
+			if (Check.isBlank(bankAccount.getName()))
+			{
+				bankAccount.setName(bankDetails);
+			}
+
+			if (Check.isBlank(bankAccount.getDescription()))
+			{
+				bankAccount.setDescription(bankDetails);
+			}
+		}
 
 		return bankAccount;
```

Comment:
- as mentioned before, we should not query by `bpartnerAccount.description`
- the way I see it, I would only use this method to locate a `C_Bank` by `bankDetails`, if one can be found, then propagate the C_Bank_ID to BPAccount